src/Workspaces/Core/Portable/ObsoleteSymbol/AbstractObsoleteSymbolService.cs

```diff
@@ -0,0 +1,176 @@
+            // locals/parameters without any references in the span, or references that don't have the declarations in 
+            // the span
+            stack.Add(root.FindNode(span));
+
+            // Use a stack so we don't blow out the stack with recursion.
+            while (stack.Count > 0)
+            {
+                var current = stack.Last();
+                stack.RemoveLast();
+
+                if (current.Span.IntersectsWith(span))
+                {
+                    var tokenFromNode = ProcessNode(semanticModel, current);
+
+                    foreach (var child in current.ChildNodesAndTokens())
+                    {
+                        if (child.IsNode)
+                        {
+                            stack.Add(child.AsNode()!);
+                        }
+
+                        var token = child.AsToken();
+                        if (token != tokenFromNode)
+                            ProcessToken(semanticModel, child.AsToken());
+
+                        foreach (var trivia in token.LeadingTrivia)
+                        {
+                            if (trivia.HasStructure)
+                            {
+                                stack.Add(trivia.GetStructure()!);
+                            }
+                        }
+
+                        foreach (var trivia in token.TrailingTrivia)
+                        {
+                            if (trivia.HasStructure)
+                            {
+                                stack.Add(trivia.GetStructure()!);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        SyntaxToken ProcessNode(SemanticModel semanticModel, SyntaxNode node)
+        {
+            if (syntaxFacts.IsUsingAliasDirective(node))
+            {
+                syntaxFacts.GetPartsOfUsingAliasDirective(node, out _, out var aliasToken, out var name);
+                if (!aliasToken.Span.IsEmpty)
+                {
+                    // Use 'name.Parent' because VB can't resolve the declared symbol directly from 'node'
+                    var symbol = semanticModel.GetDeclaredSymbol(name.GetRequiredParent(), cancellationToken);
+                    if (IsSymbolObsolete(symbol))
+                        result.Add(aliasToken.Span);
+                }
+
+                return aliasToken;
+            }
+            else if (syntaxFacts.IsImplicitObjectCreationExpression(node))
+            {
+                syntaxFacts.GetPartsOfImplicitObjectCreationExpression(node, out var creationKeyword, out _, out _);
+                if (!creationKeyword.Span.IsEmpty)
+                {
+                    var symbol = semanticModel.GetSymbolInfo(node, cancellationToken).Symbol;
+                    if (IsSymbolObsolete(symbol) || IsSymbolObsolete(symbol?.ContainingType))
```

Comment:
maybe doc.  up to you.  but i do think it would be good to potentially say "the symbol should be constructor symbol for the type, but we want to strike it out if either the constructor *or* the type is obsolete".

Reply:
➡️ Docs added, and slightly tweaked the behavior